sklearn/cluster/_kmeans.py

```diff
@@ -254,6 +255,11 @@ def _tolerance(X, tol):
     return np.mean(variances) * tol
 
 
+def _is_arraylike_not_scalar(array):
+    """Return True if array is array-like and not a scalar"""
+    return _is_arraylike(array) and not np.isscalar(array)
```

Comment:
maybe directly put this in utils.validation. It might be useful elsewhere at some point.